TypescriptProjects/jobportal/src/api/api.ts

```diff
@@ -0,0 +1,375 @@
+import axios, { AxiosResponse } from "axios";
+import {
+  SIGNUP_URL,
+  LOGIN_URL,
+  POST_JOB_URL,
+  GET_JOB_URL,
+  GET_JOB_EMPLOYER_URL,
+  GET_SINGLE_JOB_URL,
+  GET_SINGLE_JOB_EMPLOYER_URL,
+  JWT_COOKIE,
+  GET_SINGLE_JOB_APPLICATION_URL,
+  CHANGE_PASSWORD_URL,
+  BE_ENDPOINT,
+  GET_USER_URL,
+} from "../Constants";
+import { useDispatch } from "react-redux";
+import { setUserType } from "../store/store";
+
+interface UserResponse {
+  jwt: string;
+  user_type: string;
+}
+
+interface JobResponse {
+  [key: string]: any;
+}
+
+interface ApplicationResponse {
+  [key: string]: any;
+}
+
+interface ChangePasswordResponse {
+  detail: string;
+}
+
+const api = axios.create({
+  baseURL: BE_ENDPOINT,
+  headers: {
+    "Content-Type": "application/json",
+    Authorization: JWT_COOKIE,
+  },
+});
+
+export const registerUser = async (
+  fullname: string,
+  email: string,
+  password: string,
+  userType: string
+): Promise<any> => {
+  try {
+    const response: AxiosResponse<any> = await api.post(SIGNUP_URL, {
+      name: fullname,
+      email: email,
+      password: password,
+      user_type: userType,
+    });
+
+    return response.data;
+  } catch (error: any) {
+    console.log(error);
+    if (error.response && error.response.data && error.response.data.email) {
```

Comment:
@codal-kcolaco You can update this condition to `error?.response?.data?.email` same below condition as well `error?.response?.data?.password`